### ISSUE TITLE
Crash after launching any ISO on Fedora 34

### ISSUE THREAD

**Issue by lararosekelley** [project-slippi/Ishiiruka#310]
Issue tracker is **ONLY** used for reporting bugs. Please use our [discord](https://slippi.gg/discord) for support and/or feature requests.

## Expected Behavior

- Launching an ISO would open successfully

## Current Behavior

After launch, I get a black screen, and then this C/C++ error:

```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stoi
Aborted (core dumped)
```

## Steps to Reproduce

1. Fresh Fedora 34 install
2. Download Slippi AppImage
3. Download the following modules to resolve Perl errors - perl-Net-DBus perl-X11-Protocol perl-Sys-Hostname
4. Launch Slippi, then click "Play"

## Environment

Fedora 34

**Comment by tylucaskelley** [comment]
[dolphin-emu.zip](https://github.com/project-slippi/Ishiiruka/files/7003684/dolphin-emu.zip)

Attached a core dump if that helps

**Comment by lararosekelley** [comment]
Attaching core dump in case it is useful - 

[dolphin.zip](https://github.com/project-slippi/Ishiiruka/files/7003723/dolphin.zip)

**Comment by lararosekelley** [comment]
Seems like launching the ISO from standalone netplay works, but not from the Slippi launcher. Going to attempt building from source with additional logging, we have a theory on Discord that the parsing of the user.json file is busted somehow

**Comment by mitchhit234** [comment]
I was having the same issue just now, using Fedora 33 though, but the weird thing to me was that my other Fedora machine was running just fine. I went and checked each user.json file as you pointed out and it seems that the Slippi download process I just went through on the machine with the issue had generated a user.json file in a different format than my original user.json

I copied over that working user.json file to my machine and everything worked after that. If you don't have a good/bad user.json file to compare, here was each file's fields

Bad -
connectCode
playKey
uid

Good-
uid
displayName
playKey
conenctCode
latestVersion

Curious to see if this helps you with your issue!
